### PR TITLE
Replace sleep with explicit wait for authenticator state tx

### PR DESCRIPTION
**Pull request by mystenmark**

### CODE REVIEW

File: crates/sui-e2e-tests/tests/zklogin_tests.rs [hunk 1]
```diff
@@ -179,18 +180,44 @@ async fn test_create_authenticator_state_object() {
     }
 }
 
+async fn wait_for_authenticator_state_update(test_cluster: &TestCluster) {
```
Comment (lxfind) [inline crates/sui-e2e-tests/tests/zklogin_tests.rs:183]: Make it a method in test cluster?

Comment (mystenmark) [reply crates/sui-e2e-tests/tests/zklogin_tests.rs]: good call

**Comment by lxfind** [comment]
What happened to this PR?

**Comment by mystenmark** [comment]
Umm, i must have misclicked?

**Comment by mystenmark** [comment]
I must have misclicked or something?

**Comment by mystenmark** [comment]
Ok, I guess I accidentally force-pushed the wrong commit after addressing your comments

**Review by lxfind — APPROVED** [review]

**Pull request merged**